### ISSUE TITLE
Cannot read properties of undefined (reading 'split')

### ISSUE THREAD

**Issue by onesyah05** [junhoyeo/threads-api#118]
i want use like and post method but something error

TypeError: Cannot read properties of undefined (reading 'split')
    at d.<anonymous> (D:\quote_threads\node_modules\threads-api\build\threads-api.js:1:9817)
    at D:\quote_threads\node_modules\threads-api\build\threads-api.js:1:1769
    at Object.next (D:\quote_threads\node_modules\threads-api\build\threads-api.js:1:1874)
    at o (D:\quote_threads\node_modules\threads-api\build\threads-api.js:1:350)
    at i (D:\quote_threads\node_modules\threads-api\build\threads-api.js:1:547)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5

**Comment by dmytrostriletskyi** [comment]
The `Instagram API` changed their endpoint to perform a login. Either completely or any small parts like a payload and headers.

**Comment by ishaanbedi** [comment]
Having the same issue! Any progress with it @onesyah05 ?

**Comment by francistogram** [comment]
I’m pretty sure that the Threads team changed how auth works 

I made an issue before documenting what I found
https://github.com/junhoyeo/threads-api/issues/113

**Comment by aldinp16** [comment]
need to encrypt password before send login request start with #PWD:{keyId}:{time}:{ecnryptedpassword}

**Comment by francistogram** [comment]
@aldinp16 how do you get the `keyId` it seems like within the `instagram-private-api` it's [hard coded to 4](https://github.com/dilame/instagram-private-api/blob/master/src/repositories/account.repository.ts#L36)?

**Comment by aldinp16** [comment]
@francistogram try to open the apk with jadx, the key id is harcoded on apk with the public key for generate encrypted password

```java
public C95815gG() {
        this.A00 = 41;
        this.A02 = "-----BEGIN PUBLIC KEY-----\n*****-----END PUBLIC KEY-----\n";
        Calendar.getInstance().getTimeInMillis();
        this.A01 = AnonymousClass006.A00;
}
```

**Comment by junhoyeo** [comment]
@francistogram @aldinp16 Thank you guys for the research, will try to draft an quick implementation in a few hours

**Comment by junhoyeo** [comment]
almost done!

**Comment by junhoyeo** [comment]
Thanks to everyone, It was released in [v1.4.0](https://github.com/junhoyeo/threads-api/releases/tag/v1.4.0)! 👍

**Comment by cedriking** [comment]
Sorry to comment on a closed issue, but I'm still experiencing this after updating to v1.4.0. Is anyone else still experiencing the issue? 
```
ERROR (threads/64596): Cannot read properties of undefined (reading 'split')
    TypeError: Cannot read properties of undefined (reading 'split')
```
